### PR TITLE
Update `WithJsonSchema` documentation, add usage documentation for `json_schema_input_type`

### PR DESCRIPTION
**Pull request by Viicos**
## Change Summary

Fixes https://github.com/pydantic/pydantic/issues/11020.

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: docs/concepts/json_schema.md [hunk 1]
```diff
@@ -660,32 +660,24 @@ print(json.dumps(ta.json_schema(), indent=2))
...
     as it's more simple and less error-prone.
 
 The [`WithJsonSchema`][pydantic.json_schema.WithJsonSchema] annotation can be used to override the generated (base)
 JSON schema for a given type without the need to implement `__get_pydantic_core_schema__`
-or `__get_pydantic_json_schema__` on the type itself.
-
-This provides a way to set a JSON schema for types that would otherwise raise errors when producing a JSON schema,
-such as `Callable`, or types that have an [`is-instance`][pydantic_core.core_schema.is_instance_schema] core schema.
-
-For example, the use of a [`PlainValidator`][pydantic.functional_validators.PlainValidator] in the following example
```
Comment (Viicos) [inline docs/concepts/json_schema.md:674]: I couldn't find a valid example where `WithJsonSchema` can be useful with validators/serializers, as we know provide `json_schema_input_type`. Let me know if you can find one

**Review by sydney-runkle — APPROVED** [review]
Makes sense, and we've documented `json_schema_input_type` well.

**Comment by sydney-runkle** [comment]
Actually, one request before we merge - could you add a link to the `json_schema_input_type` conceptual section from here, given that they're semi-related?

**Review by sydney-runkle — APPROVED** [review]
Approved again given new conceptual docs!

**Pull request merged**